qutebrowser/browser/shared.py

```diff
@@ -462,8 +472,49 @@ def _execute_fileselect_command(
             message.error(f"Failed to open tempfile {tmpfilename} ({e})!")
             selected_files = []
 
-    if not multiple:
+    return _validated_selected_files(qb_mode=qb_mode, selected_files=selected_files)
+
+
+def _validated_selected_files(
+    qb_mode: FileSelectionMode,
+    selected_files: List[str],
+) -> List[str]:
+    """Validates selected files if they are.
+
+        * Of correct type
+        * Of correct number
+        * Existent
+
+    Args:
+        qb_mode: File selection mode used
+        selected_files: files selected
+
+    Return:
+        List of selected files that pass the checks.
+    """
+    if not qb_mode == FileSelectionMode.multiple_files:
         if len(selected_files) > 1:
-            message.warning("More than one file chosen, using only the first")
-            return selected_files[:1]
-    return selected_files
+            message.warning("More than one file/folder chosen, using only the first")
+            selected_files = selected_files[:1]
+    ignore_indices = set()
```

Comment:
I think it might be simpler to build a new `filtered_files` list and then add them to the new list if they pass the validation, instead of saving invalid indices and then removing those indices at the end - what do you think?

You could even make this a [generator](https://book.pythontips.com/en/latest/generators.html) and use `yield selected_file` if it was valid - then you don't need a list for the end result at all. Might need an additional `list(...)` around the `_validated_selected_files` call, though.

Reply:
That's indeed a lot cleaner :) Updated!